blockchain/difficulty.go

```diff
@@ -286,7 +287,8 @@ func (b *BlockChain) calcNextRequiredDifficulty(lastNode *blockNode, newBlockTim
 
 	// Limit the amount of adjustment that can occur to the previous
 	// difficulty.
-	actualTimespan := lastNode.timestamp.UnixNano() - firstNode.timestamp.UnixNano()
```

Comment:
IIRC I checked that a long time ago and no overflows would happen until 2262, which is well past the point in 2106 where the network would break without changing the timestamp in the header anyways.